cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -542,63 +536,92 @@ private void checkReconfigurability(NodeRef nodeRef, Pod pod, RestartContext res
             // and deleting a different pod in the meantime will likely result in another unschedulable pod.
             throw new FatalProblem("Pod is unschedulable or is not starting");
         }
-        // Unless the annotation is present, check the pod is at least ready.
-        boolean needsRestart = reasonToRestartPod.shouldRestart();
-        KafkaBrokerConfigurationDiff diff = null;
-        KafkaBrokerLoggingConfigurationDiff loggingDiff = null;
-        boolean needsReconfig = false;
-        // Always get the broker config. This request gets sent to that specific broker, so it's a proof that we can
-        // connect to the broker and that it's capable of responding.
-        if (!initAdminClient()) {
-            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted, because it does not seem to responding to connection attempts", nodeRef);
-            reasonToRestartPod.add(RestartReason.POD_UNRESPONSIVE);
+
+        if (podStuck) {
+            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted, because it seems to be stuck and restart might help", nodeRef);
+            restartContext.restartReasons.add(RestartReason.POD_STUCK);
             restartContext.needsRestart = false;
             restartContext.needsReconfig = false;
             restartContext.forceRestart = true;
             restartContext.diff = null;
             restartContext.logDiff = null;
             return;
         }
-        Config brokerConfig;
-        try {
-            brokerConfig = brokerConfig(nodeRef);
-        } catch (ForceableProblem e) {
-            if (restartContext.backOff.done()) {
-                needsRestart = true;
-                brokerConfig = null;
+
+        boolean adminClientInitialised = initAdminClient();
+
+        // Unless the annotation is present, check the pod is at least ready.
```

Comment:
What annotation does this refer to?

Reply:
I'm not actually sure.

Reply:
@tombentley Do you have any idea? If not, I guess we should remove it. Comment that nobody understands is useless ;-)

Reply:
Perhaps, it's talking about about manual rolling update annotation? But then the code below it does not seem to be doing anything with annotations specifically (it just checks if restart reasons list is empty or not). I agree that the comment does not add much value, so I removed it.

Reply:
No, manual rolling update anno is handled in KafkaReconciler and the KafkaRoller just gets it through the callback in restart reasons.